sdk/communication/azure-communication-phonenumbers/src/test/java/com/azure/communication/phonenumbers/PhoneNumbersIntegrationTestBase.java

```diff
@@ -43,69 +47,67 @@
                 .endpoint(communicationEndpoint)
                 .credential(new AzureKeyCredential(communicationAccessKey));
 
-        if (getTestMode() == TestMode.RECORD) {
-            List<Function<String, String>> redactors = new ArrayList<>();
-            redactors.add(data -> redact(data, JSON_PROPERTY_VALUE_REDACTION_PATTERN.matcher(data), "REDACTED"));
-            builder.addPolicy(interceptorManager.getRecordPolicy(redactors));
+        if (shouldRecord()) {
+            builder.addPolicy(getRecordPolicy());
         }
 
         return builder;
     }
 
     protected PhoneNumbersClientBuilder getClientBuilderWithConnectionString(HttpClient httpClient) {
-        if (getTestMode() == TestMode.PLAYBACK) {
-            httpClient = interceptorManager.getPlaybackClient();
-        }
-
         PhoneNumbersClientBuilder builder = new PhoneNumbersClientBuilder();
         builder
-                .httpClient(httpClient)
+                .httpClient(getHttpClient(httpClient))
+                .addPolicy(getOverrideMSUserAgentPolicy())
                 .connectionString(CONNECTION_STRING);
 
-        if (getTestMode() == TestMode.RECORD) {
-            List<Function<String, String>> redactors = new ArrayList<>();
-            redactors.add(data -> redact(data, JSON_PROPERTY_VALUE_REDACTION_PATTERN.matcher(data), "REDACTED"));
-            builder.addPolicy(interceptorManager.getRecordPolicy(redactors));
+
+        if (shouldRecord()) {
+            builder.addPolicy(getRecordPolicy());
         }
 
         return builder;
     }
 
     protected PhoneNumbersClientBuilder getClientBuilderUsingManagedIdentity(HttpClient httpClient) {
+        
         PhoneNumbersClientBuilder builder = new PhoneNumbersClientBuilder();
         builder
-                .endpoint(new CommunicationConnectionString(CONNECTION_STRING).getEndpoint())
-                .httpClient(httpClient == null ? interceptorManager.getPlaybackClient() : httpClient);
+                .httpClient(getHttpClient(httpClient))
+                .addPolicy(getOverrideMSUserAgentPolicy())
+                .endpoint(new CommunicationConnectionString(CONNECTION_STRING).getEndpoint());
 
         if (getTestMode() == TestMode.PLAYBACK) {
             builder.credential(new FakeCredentials());
         } else {
             builder.credential(new DefaultAzureCredentialBuilder().build());
         }
 
-        if (getTestMode() == TestMode.RECORD) {
-            List<Function<String, String>> redactors = new ArrayList<>();
-            redactors.add(data -> redact(data, JSON_PROPERTY_VALUE_REDACTION_PATTERN.matcher(data), "REDACTED"));
-            builder.addPolicy(interceptorManager.getRecordPolicy(redactors));
+        if (shouldRecord()) {
+            builder.addPolicy(getRecordPolicy());
         }
 
         return builder;
     }
 
+    private HttpClient getHttpClient(HttpClient httpClient) {
+        if (httpClient == null || getTestMode() == TestMode.PLAYBACK) {
```

Comment:
I see that getClientBuilderWithConnectionString used to check if it is playback mode, while getClientBuilderUsingManagedIdentity checked for null; but neither checked for both. Is there any reason for generalizing on this?

Reply:
I'm trying to cover all cases, but it seems that testing for just one of them is enough, see my reply to @lucasrsant for my thought process